src/components/templates/Auditorium/components/SectionPreviews/SectionPreviews.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useMemo } from "react";
+import { useHistory } from "react-router";
+import classNames from "classnames";
+
+import { AuditoriumSizes } from "types/auditorium";
+import { AuditoriumVenue } from "types/venues";
+
+import { useAuditoriumSections } from "hooks/auditoriumSections";
+import { useVenueId } from "hooks/useVenueId";
+
+import { SectionPreview } from "../SectionPreview";
+import { Video } from "../Video";
+
+import "./SectionPreviews.scss";
+
+export interface SectionPreviewsProps {
+  venue: AuditoriumVenue;
+}
+
+export const SectionPreviews: React.FC<SectionPreviewsProps> = ({ venue }) => {
+  const { iframeUrl } = venue;
+
+  const venueId = useVenueId();
+  const history = useHistory();
+  const { auditoriumSections } = useAuditoriumSections(venueId);
+
+  const sectionsAmount = auditoriumSections.length;
+
+  const auditoriumSize: AuditoriumSizes = useMemo(() => {
+    if (sectionsAmount <= 4) return AuditoriumSizes.SMALL;
+
+    if (sectionsAmount > 4 && sectionsAmount <= 10) {
+      return AuditoriumSizes.MEDIUM;
+    }
+
+    return AuditoriumSizes.LARGE;
+  }, [sectionsAmount]);
+
+  const sectionPreviews = auditoriumSections.map((section) => (
+    <SectionPreview
+      key={section.id}
+      section={section}
+      onClick={() =>
+        history.push(`${history.location.pathname}/section/${section.id}`)
+      }
+    />
+  ));
```

Comment:
`useMemo`

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1251/commits/75ae7353c6a7e5d30d83b9f7309d8c755ee318c1